pkg/kubelet/volumemanager/cache/actual_state_of_world_test.go

```diff
@@ -106,9 +105,7 @@ func Test_MarkVolumeAsAttached_SuppliedVolumeName_Positive_NewVolume(t *testing.
 	err := asw.MarkVolumeAsAttached(volumeName, volumeSpec, "" /* nodeName */, devicePath)
 
 	// Assert
-	if err != nil {
-		t.Fatalf("MarkVolumeAsAttached failed. Expected: <no error> Actual: <%v>", err)
-	}
+	assert.NoErrorf(t, err, "MarkVolumeAsAttached failed. Expected: <no error> Actual: <%v>", err)
```

Comment:
This (and most of the subsequent ones) will change the behavior of the test, assert.NoErrorf is a t.Errorf, not a t.Fatalf.

@mikedanese educated me the other day that the Go style guide argues against assert helpers, perhaps this is why; the exact behavior of the test is much harder to determine at a glance. 

Given the style guide recommendation, we should revert to the previous state. Very sorry to make you repeat work 😞 .

Reply:
Oh, no, in fact, it just was due to my carelessness! It's because i could not understand it very well when studying them.

I have tried to valid it with the next experiment:
```
import (
	"testing"

	"github.com/stretchr/testify/assert"
	"fmt"
)

func Test_Just_Assert_Error(t *testing.T) {
	err := Fake_error("Test_Just_Assert_Error")
	assert.NoErrorf(t, err, "Expected no error, but exist: %v", err)

	t.Logf("~~~~%s~~~~~oh, error has been found, can not log this now", "Test_Just_Assert_Error")
}

func Test_Just_Fatal(t *testing.T) {
	err := Fake_error("Test_Just_Fatal")
	if err != nil {
		t.Fatalf("Expected no error, but exist: %v", err)
	}

	t.Logf("~~~~%s~~~~~oh, oh, error has been found, can not log this now", "Test_Just_Fatal")
}

func Fake_error(name string) (err error) {
	return fmt.Errorf("Test %s error: %s", name, "error exist")
}
```
and the following will be printed:
```
--- FAIL: Test_Just_Assert_Error (0.00s)
        Error Trace:    just_assert_test.go:28
        Error:          Received unexpected error:
                        Test Test_Just_Assert_Error error: error exist
        Messages:       Expected no error, but exist: Test Test_Just_Assert_Error error: error exist
        just_assert_test.go:30: ~~~~Test_Just_Assert_Error~~~~~oh, error has been found, can not log this now
--- FAIL: Test_Just_Fatal (0.00s)
        just_assert_test.go:36: Expected no error, but exist: Test Test_Just_Fatal error: error exist
FAIL
FAIL    command-line-arguments  0.006s
```
they are really not the same between assert.NoErrorf and t.Fatalf.

I follow their source codes:
1)  assert.NoErrorf  will invoke the "t.Errorf -> t.Fail", but will not exit
2)  t.Fatalf will invoke "t.FailNow", except the t.Fail, it will invoke "runtime.Goexit()"

Of course, we have another method to express this test by assert:
```
import (
	"testing"

	"github.com/stretchr/testify/assert"
	"fmt"
)

func Test_Just_Assert_Error(t *testing.T) {
	err := Fake_error("Test_Just_Assert_Error")
	if assert.Error(t, err) {
		t.Fatalf("Expected no error, but exist: %v", err)
	}

	t.Logf("~~~~%s~~~~~oh, error has been found, can not log this now", "Test_Just_Assert_Error")
}

func Test_Just_Fatal(t *testing.T) {
	err := Fake_error("Test_Just_Fatal")
	if err != nil {
		t.Fatalf("Expected no error, but exist: %v", err)
	}

	t.Logf("~~~~%s~~~~~oh, oh, error has been found, can not log this now", "Test_Just_Fatal")
}

func Fake_error(name string) (err error) {
	return fmt.Errorf("Test %s error: %s", name, "error exist")
}
```
and the result is:
```
--- FAIL: Test_Just_Assert_Error (0.00s)
        just_assert_test.go:29: Expected no error, but exist: Test Test_Just_Assert_Error error: error exist
--- FAIL: Test_Just_Fatal (0.00s)
        just_assert_test.go:38: Expected no error, but exist: Test Test_Just_Fatal error: error exist
FAIL
FAIL    command-line-arguments  0.006s
```

Not any other advantage except the assert, so i didn't use it in this file. I hope you can give me better advice, thank you!